src/dodal/devices/bimorph_mirror.py

```diff
@@ -0,0 +1,130 @@
+import asyncio
+from typing import Annotated as A
+
+from bluesky.protocols import Movable
+from ophyd_async.core import (
+    DEFAULT_TIMEOUT,
+    AsyncStatus,
+    DeviceVector,
+    SignalR,
+    SignalRW,
+    SignalW,
+    StandardReadable,
+    StrictEnum,
+    wait_for_value,
+)
+from ophyd_async.core import StandardReadableFormat as Format
+from ophyd_async.epics.core import (
+    EpicsDevice,
+    PvSuffix,
+    epics_signal_r,
+    epics_signal_w,
+    epics_signal_x,
+)
+
+
+class BimorphMirrorOnOff(StrictEnum):
+    ON = "ON"
+    OFF = "OFF"
+
+
+class BimorphMirrorMode(StrictEnum):
+    HI = "HI"
+    NORMAL = "NORMAL"
+    FAST = "FAST"
+
+
+class BimorphMirrorStatus(StrictEnum):
+    IDLE = "Idle"
+    BUSY = "Busy"
+    ERROR = "Error"
+
+
+class BimorphMirrorChannel(StandardReadable, EpicsDevice):
+    """Collection of PVs comprising a single bimorph channel.
+
+    Attributes:
+        vtrgt: Float RW_RBV for target voltage, which can be set using parent mirror's all target proc
+        vout: Float RW_RBV for current voltage on bimorph
+        status: BimorphMirrorOnOff readable for ON/OFF status of channel
+        shift: Float writeable shifting channel voltage
+    """
+
+    vtrgt: A[SignalRW[float], PvSuffix.rbv("VTRGT"), Format.CONFIG_SIGNAL]
+    vout: A[SignalRW[float], PvSuffix.rbv("VOUT"), Format.HINTED_SIGNAL]
+    status: A[SignalR[BimorphMirrorOnOff], PvSuffix("STATUS"), Format.CONFIG_SIGNAL]
+    shift: A[SignalW[float], PvSuffix("STATUS")]
+
+
+class BimorphMirror(StandardReadable, Movable):
+    """Class to represent CAENels Bimorph Mirrors.
+
+    Attributes:
+        number_of_channels: Non-ophyd int holding number_of_channels passed into __init__
+        channels: DeviceVector of BimorphMirrorChannel, indexed from 1, for each channel
+        on_off: Writeable BimorphOnOff
+        alltrgt_proc: Procable signal that writes values in each channel's VTRGT to VOUT
+        status: Readable BimorphMirrorStatus Busy/Idle status
+        err: Alarm status"""
+
+    def __init__(self, prefix: str, number_of_channels: int, name=""):
+        """
+        Args:
+            prefix: str PV prefix
+            number_of_channels: int number of channels on bimorph mirror (can be zero)
+            name: str name of device
+
+        Raises:
+            ValueError: number_of_channels is less than zero"""
+
+        if number_of_channels < 0:
+            raise ValueError(f"Number of channels is below zero: {number_of_channels}")
```

Comment:
Could: Is there a valid use case for 0 channels? Maybe we should limit it to 1?

Reply:
There are PVs on the mirror that are not in the channel. There is a use case for accessing these without needing access to the channels.

Reply:
Added test with zero channels.